modules/solid_mechanics/include/bcs/DirectDirichletBCBase.h

```diff
@@ -0,0 +1,62 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "NodalBC.h"
+#include "libmesh/libmesh_common.h"
+#include "libmesh/sparse_matrix.h"
+#include "libmesh/numeric_vector.h"
+
+namespace libMesh
+{
+template <typename T>
+class NumericVector;
+class NonlinearImplicitSystem;
+} // namespace libMesh
+
+// libMesh forward declarations
+namespace libMesh
+{
+template <typename T>
+class SparseMatrix;
+}
+/**
+ * Base boundary condition of a Dirichlet type
+ */
+class DirectDirichletBCBase : public NodalBC
+{
+public:
+  static InputParameters validParams();
+
+  DirectDirichletBCBase(const InputParameters & parameters);
+
+  virtual void initialSetup() override;
+
+  void computeValue(NumericVector<Number> & current_solution);
+
+  bool preset() const { return _preset; }
```

Comment:
It would if your class was derived from `DirichletBCBase` but your class is derived directly from `NodalBC`. Another way to think about it is that since this function is not an overridden virtual it is impossible for it to be called from the framework

Reply:
[ab8b9fe](https://github.com/idaholab/moose/pull/28175/commits/ab8b9fe85692cb185dda7f6264efbf1c75a0e597)